kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsLocal.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.installation.kms.aws;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.ServicePort;
+
+import io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+import io.kroxylicious.systemtests.resources.manager.ResourceManager;
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+import io.kroxylicious.systemtests.utils.TestUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * The type Aws kms local.
+ */
+public class AwsKmsLocal implements AwsKmsClient {
+    public static final String LOCALSTACK_SERVICE_NAME = "localstack";
+    public static final String LOCALSTACK_DEFAULT_NAMESPACE = "localstack";
+    public static final String LOCALSTACK_HELM_REPOSITORY_URL = "https://localstack.github.io/helm-charts";
+    public static final String LOCALSTACK_HELM_REPOSITORY_NAME = "localstack";
+    public static final String LOCALSTACK_HELM_CHART_NAME = "localstack/localstack";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsKmsLocal.class);
+    private static final String AWS_LOCAL_CMD = "awslocal";
+    private final String deploymentNamespace;
+    private String podName;
+    private String installedLocalStackVersion;
+    private String region;
+
+    /**
+     * Instantiates a new Aws.
+     *
+     */
+    public AwsKmsLocal() {
+        this.deploymentNamespace = LOCALSTACK_DEFAULT_NAMESPACE;
+    }
+
+    @Override
+    public String getAwsCmd() {
+        return AWS_LOCAL_CMD;
+    }
+
+    /**
+     * Is deployed
+     *
+     * @return true if Aws service is deployed in kubernetes, false otherwise
+     */
+    public boolean isDeployed() {
+        return kubeClient().getService(deploymentNamespace, LOCALSTACK_SERVICE_NAME) != null;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return Environment.AWS_ACCESS_KEY_ID.equals(Environment.AWS_ACCESS_KEY_ID_DEFAULT);
```

Comment:
I was confused as well, but after looking at the `TestKmsFacadeInvocationContextProvider` from the integration tests, it is not a health check, it is the ability to access to the corresponding `TestKmsFacade`.
In other classes from the integration tests, the only thing done is checking that Docker is available.

Reply:
It is supposed to indicate if the facade was available or not.  In the integration tests, we use it to skip the tests that require Docker/Podman if it is not available.

```java
/**
     * Returns true of this facade is available, or false otherwise.
     * @return true if available, false otherwise.
     */
```